satpy/readers/yaml_reader.py

```diff
@@ -1154,8 +1155,28 @@ class GEOSegmentYAMLReader(GEOFlippableFileYAMLReader):
     field which will be used if ``expected_segments`` is not defined. This
     will default to 1 segment.
 
+    This reader uses an optional ``preload`` keyword argument that can be
+    passed to the :meth:`~satpy.scene.Scene` class upon instantiation.
+    This argument is intended for near real time processing.  When only
+    one segment has arrived, the user can create a scene using this one
+    segment and ``preload=True``, and Satpy will populate a scene based on
+    all expected segments with dask delayed objects.  When computing the
+    dask graphs, satpy will process each segment as soon as it comes in,
+    strongly reducing the timeliness for processing a full disc image in
+    near real time.  Checking for new files can be controlled with the
+    arguments ``preload_step`` (time in seconds) and ``preload_tries``
+    (how many tries before giving up).
+
+    This feature is experimental.  Use at your own risk.
+
+    .. versionadded:: 0.47
     """
 
+    def __init__(self, *args, **kwargs):
+        """Initialise object."""
+        self.preload = kwargs.pop("preload", False)
+        super().__init__(*args, **kwargs)
```

Comment:
I'm wondering if we should use satpy's config system for this instead of adding yet another keyword argument. In my mind, a config would make sense as I don't think we would want to mix preloadable and non preloadable data.

Reply:
We can use a configuration variable, so that the user no longer needs to pass a new keyword argument, and I'm working on that now.  However, internally, we still need to tell the difference between a FileHandler referring to a real existing file (such as the first one of a RC) or the ones referring to a glob pattern for a file still to be created.

Reply:
Implemented in 38431d382.  The `GeoSegmentFileYAMLReader` reads a configuration parameter.  The `FileHandler` still needs a keyword argument, because even when we use preloading, the file handler corresponding to the first file is not preloaded, so we need to disambiguate between the two.